### ISSUE TITLE
Building errors

### ISSUE THREAD

**Issue by ispysoftware** [webrtc-uwp/webrtc-uwp-sdk#87]
I've cloned the repo, run the build script, opened the solution and am trying to build org.webrtc project

Getting over 100 errors, for example (despite what it says, Visual Studio 2017 is most definitely installed)

> Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0006	Metadata file 'D:\WebRTCUWP\webrtc\windows\solutions\Build\Output\Org.WebRtc\Debug\x86\Org.WebRtc.winmd' could not be found	PeerConnectionClient.WebRtc	D:\WebRTCUWP\common\windows\samples\PeerCC\Client\CSC	1	Active
Error		Cannot resolve Assembly or Windows Metadata file 'D:\WebRTCUWP\webrtc\windows\solutions\Build\Output\Org.WebRtc\Debug\x86\Org.WebRtc.winmd'	PeerConnectionClient.WebRtc	D:\WebRTCUWP\common\windows\samples\PeerCC\Client\PeerConnectionClient.WebRtc.csproj		
Error		Visual Studio 2017 is not installed[0m	WebRtc.UWP.Native.Builder	D:\WebRTCUWP\webrtc\windows\projects\msvc\WebRtc.UWP.Native.Builder\EXEC	1	
Error	MSB3073	The command "pushd "D:\WebRTCUWP\webrtc\windows\projects\msvc\WebRtc.UWP.Native.Builder\..\..\..\..\..\" && call bin\prepare.bat -logLevel 4 -platform winuwp -cpu x86 -config Debug -target webrtc && call bin\buildWebRTC.bat Debug winuwp x86 webrtc && popd
:VCEnd" exited with code 255.	WebRtc.UWP.Native.Builder	d:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets	123	
Error	C1083	Cannot open include file: 'generated/types.h': No such file or directory (compiling source file ..\..\..\..\xplatform\webrtc\sdk\windows\wrapper\impl_org_webRtc_EventQueueMaker.cpp)	Org.WebRtc.WrapperGlue	d:\webrtcuwp\webrtc\xplatform\webrtc\sdk\windows\wrapper\types.h	4	
Error	C1083	Cannot open include file: 'generated/types.h': No such file or directory (compiling source file ..\..\..\..\xplatform\webrtc\sdk\windows\wrapper\impl_org_webRtc_Constraint.cpp)	Org.WebRtc.WrapperGlue	d:\webrtcuwp\webrtc\xplatform\webrtc\sdk\windows\wrapper\types.h	4	

... etc. What am I doing wrong here?

**Comment by robin-raymond** [comment]
@SergioOpenPeer is there something about drive D that is failing? Looks like the generated output did not happen.
@ispysoftware What branch are you attempting to build?

**Comment by SergioOpenPeer** [comment]
If VS is not installed in Program Files or Program Files (86), then it will show that error.
To overcome that problem it is necessary to manually set msVS_Path value in buildWebRTC.bat to proper location of Microsoft Visual Studio\2017

**Comment by robin-raymond** [comment]
@SergioOpenPeer needs better error message and resolution steps in new version of scripts.

**Comment by SergioOpenPeer** [comment]
@robin-raymond Agree.

**Comment by ispysoftware** [comment]
@robin-raymond  I just followed the instructions here:
https://github.com/webrtc-uwp/webrtc-uwp-sdk

There were some errors when running prepare.bat but nothing that stopped the build process from completing:

D:\WebRTCUWP>bin\prepare.bat
Running script with default parameters:
Target: all (Ortc and WebRtc)
Platform: all (winuwp, win32)
Cpu: all (x64, x86, arm)
Log level: 2 (warning)

Running prepare script ...

Preparing webRTC development environment ...
Preparing development environment for WinUWP and Win32 platforms ...
Preparing development environment for arm, x86, and x64 cpus ...
Preparing development environment for debug and release configurations ...
Pip and pywin32 setup...
Collecting pip
  Using cached https://files.pythonhosted.org/packages/c2/d7/90f34cb0d83a6c5631cf71dfe64cc1054598c843a92b400e55675cc2ac37/pip-18.1-py2.py3-none-any.whl
Collecting setuptools
  Downloading https://files.pythonhosted.org/packages/e7/16/da8cb8046149d50940c6110310983abb359bbb8cbc3539e6bef95c29428a/setuptools-40.6.2-py2.py3-none-any.whl (573kB)
    100% |################################| 573kB 3.9MB/s
Collecting wheel
  Downloading https://files.pythonhosted.org/packages/5a/9b/6aebe9e2636d35d1a93772fa644c828303e1d5d124e8a88f156f42ac4b87/wheel-0.32.2-py2.py3-none-any.whl
Installing collected packages: pip, setuptools, wheel
Successfully installed pip-18.1 setuptools-40.6.2 wheel-0.32.2
Requirement already up-to-date: pip in c:\python27\lib\site-packages (18.1)
Collecting pywin32
  Downloading https://files.pythonhosted.org/packages/fc/7c/25f36684ec3b8ea47d5601442950a4fa770d066f57171fbf8c905ccbfd25/pywin32-224-cp27-cp27m-win32.whl (6.8MB)
    100% |################################| 6.8MB 2.6MB/s
Installing collected packages: pywin32
Successfully installed pywin32-224
        1 file(s) copied.
COPY webrtc\xplatform\templates\gn\idl_BUILD.gn webrtc\xplatform\webrtc\third_party\idl\BUILD.gn
Running WebRTC prepare script ...
=================================
Preparing WebRTC development environment for WinUWP and win32 platforms ...
Preparing WebRTC development environment for arm, x86, x64 cpus ...
Preparing WebRTC development environment for debug and release configurations ...
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
        1 file(s) copied.
In case pop-up window 'Git Credential Manager for Windows' appears after clang download, just click 'Cancel' button.
Traceback (most recent call last):
  File "tools\clang\scripts\update.py", line 931, in <module>
    sys.exit(main())
  File "tools\clang\scripts\update.py", line 927, in main
    return UpdateClang(args)
  File "tools\clang\scripts\update.py", line 464, in UpdateClang
    WriteStampFile('')
  File "tools\clang\scripts\update.py", line 168, in WriteStampFile
    with open(path, 'w') as f:
IOError: [Errno 2] No such file or directory: 'D:\\WebRTCUWP\\webrtc\\xplatform\\webrtc\\third_party\\llvm-build\\cr_build_revision'
0> Downloading buildtools\win\gn.exe...
Downloading 1 files took 116.434000 second(s)
0> Downloading buildtools\win\clang-format.exe...
Downloading 1 files took 5.762000 second(s)
Generating WebRTC WinUWP projects for x86 cpu ...
Generating WebRTC Win32 projects for x86 cpu ...
Generating WebRTC WinUWP projects for x64 cpu ...
Generating WebRTC Win32 projects for x64 cpu ...
Generating WebRTC WinUWP projects for arm cpu ...

Success: WebRTC development environment is prepared.
=======   WebRTC prepare script summary   =======
=======   platform   =========   cpu   =========   result   =======
          winuwp                 arm               prepared
          winuwp                 x86               prepared
          winuwp                 x64               prepared
          win32                  x86               prepared
          win32                  x64               prepared
===================================================================


Success: Development environment is set.
Total execution time: 00:05:59 (359s total)

**Comment by ispysoftware** [comment]
I started over, updated VS 2017 to the latest version, git clone command completed without errors, bin\prepare.bat completed without errors. I set the program files path to point to the correct location on my d drive in buildWebRTC.bat

Still getting over 100 errors trying to build the generated solution. The first error is

The command "pushd "D:\webrtcuwp\webrtc-uwp-sdk\webrtc\windows\projects\msvc\WebRtc.UWP.Native.Builder\..\..\..\..\..\" && call bin\prepare.bat -logLevel 4 -platform winuwp -cpu x86 -config Debug -target webrtc && call bin\buildWebRTC.bat Debug winuwp x86 webrtc && popd
:VCEnd" exited with code 255.
